pkg/k8smgmt/cluster.go

```diff
@@ -52,12 +55,22 @@ func SetMasterNoscheduleTaint(ctx context.Context, client ssh.Client, masterName
 	log.SpanLog(ctx, log.DebugLevelInfra, "SetMasterNoscheduleTaint", "masterName", masterName, "action", action)
 
 	var cmd string
+	taintAnnotation := NoScheduleMasterTaintAnnotaionNew
+	cmd = fmt.Sprintf("kubectl version --short --kubeconfig=%s | grep Server", kubeconfig)
+	out, err := client.Output(cmd)
+	if err != nil {
+		return fmt.Errorf("Unable to get k8s version, %v", err)
+	}
+	// For old clusters NoSchedule taint is different
+	if out == "Server Version: v1.18.20" {
```

Comment:
Not sure if you tested this against a v1.18 cluster but I might recommend using strings.Contains() to allow for `\n` and other extra stuff that might show up in the output.

Reply:
Good idea. I didn't test it out against 1.18 just because we don't have one that'd be easy to test.